src/txn/txn_recover.c

```diff
@@ -647,6 +627,63 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
         WT_ERR(ret);
     }
 
+    /*
+     * We should check whether the history store file exists in the metadata or not. If it does not,
+     * then we should skip rollback to stable for each table. This might happen if we're upgrading
+     * from an older version. If it does exist in the metadata we should check that it exists on
+     * disk to confirm that it wasn't deleted between runs.
+     *
+     * This needs to happen after we apply the logs as they may contain the metadata changes which
+     * include the history store creation. As such the on disk metadata file won't contain the
+     * history store but will after log application.
+     */
+    metac->set_key(metac, WT_HS_URI);
+    WT_ERR_NOTFOUND_OK(metac->search(metac), true);
+    if (ret == WT_NOTFOUND) {
+        hs_exists = false;
+    } else {
+        /* Given the history store exists in the metadata validate whether it exists on disk. */
+        WT_ERR(__wt_fs_exist(session, WT_HS_FILE, &hs_exists));
+        if (hs_exists) {
+            /* If found, save the oldest start timestamp in the checkpoint list of the history
+             * store. This will be the minimum valid oldest timestamp at restart. Its possible this
+             * will detect corruption if it fails, it can also return not found in the instance
+             * where the timestamp doesn't exist so we ignore that.
+             */
+            ret = __wt_meta_get_oldest_ckpt_timestamp(session, WT_HS_URI, &oldest_ts);
```

Comment:
@tammybailey this is where I merged your change with mine, let me know if it makes sense.